tutorial/sample2/webgl-demo.js

```diff
@@ -69,82 +64,111 @@ function initWebGL() {
 // Initialize the buffers we'll need. For this demo, we just have
 // one object -- a simple two-dimensional square.
 //
-function initBuffers() {
+function initBuffers(gl) {
 
-  // Create a buffer for the square's vertices.
+  // Create a buffer for the square's positions.
 
-  squareVerticesBuffer = gl.createBuffer();
+  const positionBuffer = gl.createBuffer();
 
-  // Select the squareVerticesBuffer as the one to apply vertex
+  // Select the positionBuffer as the one to apply buffer
   // operations to from here out.
 
-  gl.bindBuffer(gl.ARRAY_BUFFER, squareVerticesBuffer);
+  gl.bindBuffer(gl.ARRAY_BUFFER, positionBuffer);
 
-  // Now create an array of vertices for the square. Note that the Z
-  // coordinate is always 0 here.
+  // Now create an array of positions for the square.
 
-  var vertices = [
-    1.0,  1.0,  0.0,
-    -1.0, 1.0,  0.0,
-    1.0,  -1.0, 0.0,
-    -1.0, -1.0, 0.0
+  const positions = [
+     1.0,  1.0,
+    -1.0,  1.0,
+     1.0, -1.0,
+    -1.0, -1.0,
   ];
 
-  // Now pass the list of vertices into WebGL to build the shape. We
+  // Now pass the list of positions into WebGL to build the shape. We
   // do this by creating a Float32Array from the JavaScript array,
-  // then use it to fill the current vertex buffer.
+  // then use it to fill the current buffer.
 
-  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(vertices), gl.STATIC_DRAW);
+  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(positions), gl.STATIC_DRAW);
+
+  return {
+    position: positionBuffer,
+  };
 }
 
 //
-// drawScene
-//
 // Draw the scene.
 //
-function drawScene() {
+function drawScene(gl, shaderProgram, attribLocations, uniformLocations, buffers) {
+  gl.clearColor(0.0, 0.0, 0.0, 1.0);  // Clear to black, fully opaque
+  gl.clearDepth(1.0);                 // Clear everything
+  gl.enable(gl.DEPTH_TEST);           // Enable depth testing
+  gl.depthFunc(gl.LEQUAL);            // Near things obscure far things
+
   // Clear the canvas before we start drawing on it.
 
   gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
 
   // Establish the perspective with which we want to view the
   // scene. Our field of view is 45 degrees, with a width/height
-  // ratio of 640:480, and we only want to see objects between 0.1 units
+  // ratio that matches the display size of the canvas
+  // and we only want to see objects between 0.1 units
   // and 100 units away from the camera.
 
-  perspectiveMatrix = makePerspective(45, 640.0/480.0, 0.1, 100.0);
+  const fieldOfView = 45 * Math.PI / 180;   // in radians
+  const aspect = gl.canvas.clientWidth / gl.canvas.clientHeight;
+  const zNear = 0.1;
+  const zFar = 100.0;
+  const projectionMatrix = mat4.create();
+  mat4.perspective(projectionMatrix, fieldOfView, aspect, zNear, zFar);
```

Comment:
I feel like `gl-matrix` may need a small preamble before its first use. I'm not sure where this tutorial is linked from MDN, but `gl-matrix` does a few magical things. I think it would be helpful to mention how it is optimized to not allocate any objects, so the first parameter is what the output is set to.

I think a few short sentences explaining what a perspective matrix is could be helpful too. It's beyond the scope of this tutorial to fully explain it, but something along the lines of...

> Create a perspective matrix, a special matrix that is used to simulate the distortion of perspective in a camera.